src/bundle/ui-dev/src/modules/universal-discovery/components/selected-items/selected.items.panel.js

```
@@ -0,0 +1,150 @@
+    };
+    const renderToggleBtn = () => {
+        return (
+            <button
+                type="button"
+                className="c-selected-items-panel__toggle-button btn ibexa-btn ibexa-btn--tertiary ibexa-btn--small ibexa-btn--no-text"
+                onClick={toggleExpanded}
+                title={togglerLabel}
+                data-tooltip-container-selector=".c-udw-tab"
+            >
+                <Icon name="expand-left" extraClasses="c-selected-items-panel__toggle-button-icon ibexa-icon--tiny-small" />
+            </button>
+        );
+    };
+    const renderActionBtns = () => {
+        const removeLabel = Translator.transChoice(
+            /*@Desc("{1}Deselect|[2,Inf]Deselect all")*/ 'selected_items.deselect_all',
+            selectedItems.length,
+            {},
+            'ibexa_universal_discovery_widget',
+        );
+
+        return (
+            <div className="c-selected-items-panel__actions">
+                <button
+                    type="button"
+                    className="c-selected-items-panel__clear-selection-button btn ibexa-btn ibexa-btn--small ibexa-btn--secondary"
+                    onClick={clearSelection}
+                >
+                    {removeLabel}
+                </button>
+            </div>
+        );
+    };
+    const renderLocationsList = () => {
+        if (!isExpanded) {
+            return null;
+        }
+
+        return (
+            <div className="c-selected-items-panel__items-wrapper">
+                {renderActionBtns()}
+                <div className="c-selected-items-panel__items-list">
+                    {selectedItems.map((selectedItem) => {
+                        const ItemComponent = itemsComponentsMap[selectedItem.type].component;
+
+                        return ItemComponent && <ItemComponent key={`${selectedItem.type}-${selectedItem.id}`} item={selectedItem} />;
```

is there a chance component wouldn't exist? and if there is, do we want to silent error the fact there's probably some misconfiguration and specific selected item doesn't show up?

---

good point!